envi/archs/i386/disasm.py

```diff
@@ -122,7 +122,7 @@
     opconst.INS_XCHGCC: envi.IF_COND,
     opconst.INS_TRET: envi.IF_NOFALL | envi.IF_RET,
     opconst.INS_HALT: envi.IF_NOFALL,
-    opconst.INS_TRAP: envi.IF_NOFALL,
+    # opconst.INS_TRAP: envi.IF_NOFALL,
```

Comment:
I'm not against it, but I felt like a more general approach was warranted. See my other comment.

Reply:
i'm usually all for the ArchitectureModule handling specifics for the architecture, so adding callbacks like this are totally good with me.

however, for performance-sake, i very much want IF_INTERRUPT or something like it, so that we're only calling another callback on instructions that warrant it.  

unfortunately, because it's a performance issue and gets run against every single instruction decode/emulation, this is very important to me.